pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller_test.go

```diff
@@ -0,0 +1,242 @@
+			objs := []runtime.Object{&node}
+			cl := fake.NewFakeClient(objs...)
+			predicate := forThisNodePredicate(cl)
+
+			Expect(predicate.
+				CreateFunc(event.CreateEvent{
+					Object: &nodeNetworkConfigurationPolicy})).To(Equal(c.Reconcile))
+			Expect(predicate.
+				DeleteFunc(event.DeleteEvent{
+					Object: &nodeNetworkConfigurationPolicy})).To(Equal(c.Reconcile))
+			Expect(predicate.
+				GenericFunc(event.GenericEvent{
+					Object: &nodeNetworkConfigurationPolicy})).To(Equal(c.Reconcile))
+			Expect(predicate.
+				UpdateFunc(event.UpdateEvent{
+					ObjectOld: &nodeNetworkConfigurationPolicy,
+					ObjectNew: &nodeNetworkConfigurationPolicy,
+				})).To(Equal(c.Reconcile))
+		},
+		Entry("events with empty node labels",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels:  map[string]string{},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: false,
+			}),
+		Entry("events with nil node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: nil,
+				Reconcile:    true,
+			}),
+		Entry("events with empty node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: map[string]string{},
+				Reconcile:    true,
+			}),
+		Entry("events with matching node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: true,
+			}),
+		Entry("events with missing label at node",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: false,
+			}),
+		Entry("events with different label value at node",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar1",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: false,
+			}),
+		Entry("node not found",
+			predicateCase{
+				ObjNodeName:  "node02",
+				NodeLabels:   map[string]string{},
+				NodeSelector: map[string]string{},
+				Reconcile:    false,
+			}),
+	)
+})
+
+var _ = Describe("NodeNetworkConfigurationPolicy controller reconciler", func() {
+	var (
+		name          = "nodenetworkconfigurationpolicy-test"
+		namespace     = "default"
+		numberOfNodes = 2
+		nodes         []nmstatev1alpha1.NodeNetworkState
+
+		cl      client.Client
+		r       *ReconcileNodeNetworkConfigurationPolicy
+		req     reconcile.Request
+		policy1 = nmstatev1alpha1.NodeNetworkConfigurationPolicy{
```

Comment:
policy1? what's that

Reply:
This is the policy that we reconcile.

Reply:
Then it should be reconciledPolicy (or something like that). Also, it shoudn't be a global variable - you need to make sure it's clean before ever other test.